src/metabase/upload.clj

```diff
@@ -606,7 +606,7 @@
         (ex-info (tru "The table must be an uploaded table.")
                  {:status-code 422})
 
-        (not (mi/can-write? table))
```

Comment:
I had this wrong before. You can read more about table permissions here:
https://github.com/metabase/metabase/blob/1bc18575ed8e453c42024a077d58ac96ca4cf0e5/src/metabase/models/table.clj#L70-L79

Reply:
from the doc 
> unrestricted data permissions on the table

could you explain how `can-write` resonates with that?

Reply:
It doesn't. That's why I changed it from `can-write?` to `can-read?`.

And from the comments on table permissions I linked above:
[1bc1857/src/metabase/models/table.clj#L70-L79](https://github.com/metabase/metabase/blob/1bc18575ed8e453c42024a077d58ac96ca4cf0e5/src/metabase/models/table.clj#L70-L79)
  ;; To read (e.g., fetch metadata) a Table you must have either self-service data permissions for the Table, or write
  ;; permissions for the Table (detailed below). `can-read?` checks the former, while `can-write?` checks the latter;